### PR TITLE
Feature/todoapp scaffold

### PR DESCRIPTION
**Pull request by nikolay-vasilev-prime**
#### Overview

Todo application example with rx_bloc and rx_bloc_cli.

#### Checklist

*Implementation*
- [x] Implementation matches ticket acceptance criteria and technical notes
- [x] Manually tested against Acceptance Criteria
- [x] UI checked in Light / Dark mode

*Stability*
- [x] Checked if changes affect any features and verified affected features work as expected
- [x] Added unit tests for new code and verified existing tests work as expected

*Code quality*
- [x] Updated `CHANGELOG.md`, `README.md` and package versions in `pubspec.yaml`
- [x] Dependencies are updated to latest versions or new tickets are created if there are breaking changes or deprecations
- [x] If an unrelated part of the codebase needs to be updated or refactored, create tickets with proposed changes

### CODE REVIEW

**Review by StanevPrime — CHANGES_REQUESTED** [review]
The todo app tests need to be automatically executed and integrated into the existing CI pipeline

**Review by JvnSlv — APPROVED** [review]

File: .github/workflows/build_and_test_rx_bloc_packages.yaml [hunk 1]
```diff
@@ -50,7 +51,7 @@ jobs:
           fi
       - name: Remove unnecessary files
         working-directory: ${{ matrix.folder }}
-        run: flutter pub run clean_coverage clean --exclusions '**/.g.dart','**repository.dart','**rxb.g.dart' coverage/lcov.info
+        run: flutter pub run clean_coverage clean --exclusions '**/.g.dart','**/di/**','**repository.dart','**/models/**','**/routes.dart','**data_source.dart','**_model.g.dart','**_models.g.dart''**_model.dart','**_model_extensions.dart','**_models.dart','**_models_extensions.dart','**_config.dart','**_constants.dart','**_translations.dart','lib/assets.dart','lib/keys.dart','lib/base/theme/**','lib/base/utils/**','lib/keys.dart','lib/l10n/**' coverage/lcov.info
```
Comment (StanevPrime) [inline .github/workflows/build_and_test_rx_bloc_packages.yaml:54]: @DDavidPrime do you think we need to apply these changes in the rx_bloc_cli ?

File: examples/todoapp/bin/server/controllers/translations_controller.dart [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+// Use of this source code is governed by an MIT-style
+// license that can be found in the LICENSE file or at
+// https://opensource.org/licenses/MIT.
+
+import 'package:shelf/shelf.dart';
+
+import '../repositories/translations_repository.dart';
+import '../utils/api_controller.dart';
+
+class TranslationsController extends ApiController {
```
Comment (StanevPrime) [inline examples/todoapp/bin/server/controllers/translations_controller.dart:13]: @DDavidPrime don't we have a translations controller in the rx_bloc_cli ?

File: packages/rx_bloc_cli/mason_templates/bricks/feature_cicd_fastlane/__brick__/{{#cicd_github}}.github{{/cicd_github}}/workflows/fastlane_android_custom_build_and_deploy.yaml [hunk 3]
```diff
@@ -53,6 +53,9 @@ jobs:
         with:
           flutter-version: ${{ inputs.flutter_version }}
           channel: "stable"
+          cache: true
```
Comment (StanevPrime) [inline packages/rx_bloc_cli/mason_templates/bricks/feature_cicd_fastlane/__brick__/{{#cicd_github}}.github{{/cicd_github}}/workflows/fastlane_android_custom_build_and_deploy.yaml:56]: Please avoid any non-todoapp related changes

**Review by StanevPrime — CHANGES_REQUESTED** [review]

File: packages/rx_bloc_cli/mason_templates/bricks/feature_cicd_fastlane/__brick__/{{#cicd_github}}.github{{/cicd_github}}/workflows/fastlane_android_custom_build_and_deploy.yaml (hunk 3 above)
Comment (nikolay-vasilev-prime) [reply packages/rx_bloc_cli/mason_templates/bricks/feature_cicd_fastlane/__brick__/{{#cicd_github}}.github{{/cicd_github}}/workflows/fastlane_android_custom_build_and_deploy.yaml:56]: As long as I see it's coming from here: https://github.com/Prime-Holding/rx_bloc/blob/feat/remove-credentials-after-build-deploy/.github/workflows/fastlane_android_custom_build_and_deploy.yaml
@JvnSlv is it possible that this is from the division app fixes?

File: examples/todoapp/bin/server/controllers/translations_controller.dart (hunk 2 above)
Comment (nikolay-vasilev-prime) [reply examples/todoapp/bin/server/controllers/translations_controller.dart:13]: Is this rx_bloc_cli generated code? https://github.com/Prime-Holding/rx_bloc/blob/7a44e708a85fad6148622f3b69fcc2091872149c/examples/todoapp/bin/server/controllers/translations_controller.dart
It looks like the initial scaffold creation.

File: packages/rx_bloc_cli/mason_templates/bricks/feature_cicd_fastlane/__brick__/{{#cicd_github}}.github{{/cicd_github}}/workflows/fastlane_android_custom_build_and_deploy.yaml (hunk 3 above)
Comment (StanevPrime) [reply packages/rx_bloc_cli/mason_templates/bricks/feature_cicd_fastlane/__brick__/{{#cicd_github}}.github{{/cicd_github}}/workflows/fastlane_android_custom_build_and_deploy.yaml:56]: @DDavidPrime please review this change

Comment (JvnSlv) [reply packages/rx_bloc_cli/mason_templates/bricks/feature_cicd_fastlane/__brick__/{{#cicd_github}}.github{{/cicd_github}}/workflows/fastlane_android_custom_build_and_deploy.yaml:56]: > As long as I see, it's coming from here: https://github.com/Prime-Holding/rx_bloc/blob/feat/remove-credentials-after-build-deploy/.github/workflows/fastlane_android_custom_build_and_deploy.yaml @JvnSlv is it possible that this is from the division app fixes?

I don't think so. Only changes that I have done were related to the example projects, I didn't change anything rx_bloc_cli related

File: examples/todoapp/lib/base/common_services/deep_link_service.dart [hunk 4]
```diff
@@ -0,0 +1,26 @@
...
+// https://www.primeholding.com
+//
+// Use of this source code is governed by an MIT-style
+// license that can be found in the LICENSE file or at
+// https://opensource.org/licenses/MIT.
+
+import '../models/deep_link_model.dart';
+import '../repositories/deep_link_repository.dart';
+
+class DeepLinkService {
```
Comment (StanevPrime) [inline examples/todoapp/lib/base/common_services/deep_link_service.dart:11]: Please remove the not used classes

File: examples/todoapp/lib/base/data_sources/local/profile_local_data_source.dart [hunk 5]
```diff
@@ -0,0 +1,30 @@
...
+//
+// Use of this source code is governed by an MIT-style
+// license that can be found in the LICENSE file or at
+// https://opensource.org/licenses/MIT.
+
+import 'shared_preferences_instance.dart';
+
+/// This class is used to save user profile related settings into shared preferences
+/// Currently only notification settings are saved
+class ProfileLocalDataSource {
```
Comment (StanevPrime) [inline examples/todoapp/lib/base/data_sources/local/profile_local_data_source.dart:12]: Please remove the not used classes.

**Review by StanevPrime — CHANGES_REQUESTED** [review]

File: .github/workflows/build_and_test_rx_bloc_packages.yaml (hunk 1 above)
Comment (DDavidPrime) [inline .github/workflows/build_and_test_rx_bloc_packages.yaml:54]: @StanevPrime I've removed your comment thinking that I would remove mine by mistake. You've asked if we can include this into the rx_bloc_cli package, and I can say to that yes, we can. Will add it in a separate PR along with some other smaller changes.

File: packages/rx_bloc_cli/mason_templates/bricks/feature_cicd_fastlane/__brick__/{{#cicd_github}}.github{{/cicd_github}}/workflows/fastlane_android_custom_build_and_deploy.yaml [hunk 6]
```diff
@@ -53,9 +53,6 @@ jobs:
         with:
           flutter-version: ${{ inputs.flutter_version }}
           channel: "stable"
-          cache: true
-          cache-key: "flutter-:os:-:channel:-:version:-:arch:-:hash:-${{ hashFiles('**/pubspec.lock') }}"
-          cache-path: "${{ runner.tool_cache }}/flutter/:channel:-:version:-:arch:"
```
Comment (DDavidPrime) [inline packages/rx_bloc_cli/mason_templates/bricks/feature_cicd_fastlane/__brick__/{{#cicd_github}}.github{{/cicd_github}}/workflows/fastlane_android_custom_build_and_deploy.yaml]: Please do not remove any rx_bloc_cli template code unless there is a reason for it.

File: .github/workflows/build_and_test_rx_bloc_packages.yaml (hunk 1 above)
Comment (StanevPrime) [reply .github/workflows/build_and_test_rx_bloc_packages.yaml:54]: Ok, please proceed. The end goal is to have code coverage report only for the classes that can be tested.

File: packages/rx_bloc_cli/mason_templates/bricks/feature_cicd_fastlane/__brick__/{{#cicd_github}}.github{{/cicd_github}}/workflows/fastlane_ios_custom_build_and_deploy.yaml [hunk 7]
```diff
@@ -53,9 +53,6 @@ jobs:
         with:
           flutter-version: ${{ inputs.flutter_version }}
           channel: "stable"
-          cache: true
-          cache-key: "flutter-:os:-:channel:-:version:-:arch:-:hash:-${{ hashFiles('**/pubspec.lock') }}"
-          cache-path: "${{ runner.tool_cache }}/flutter/:channel:-:version:-:arch:"
```
Comment (DDavidPrime) [inline packages/rx_bloc_cli/mason_templates/bricks/feature_cicd_fastlane/__brick__/{{#cicd_github}}.github{{/cicd_github}}/workflows/fastlane_ios_custom_build_and_deploy.yaml]: Please do not remove any rx_bloc_cli template code unless there is a reason for it. Also make sure to always double check what files you are modifying. You're modifying rx_bloc_cli template files in a PR that has to do with the todo example app.

File: packages/rx_bloc_cli/mason_templates/bricks/feature_cicd_fastlane/__brick__/{{#cicd_github}}.github{{/cicd_github}}/workflows/fastlane_android_custom_build_and_deploy.yaml (hunk 6 above)
Comment (DDavidPrime) [reply packages/rx_bloc_cli/mason_templates/bricks/feature_cicd_fastlane/__brick__/{{#cicd_github}}.github{{/cicd_github}}/workflows/fastlane_android_custom_build_and_deploy.yaml]: After some look at the code history, it turns out this came from develop. Those changes were made intentionally so closing this.

File: packages/rx_bloc_cli/mason_templates/bricks/feature_cicd_fastlane/__brick__/{{#cicd_github}}.github{{/cicd_github}}/workflows/fastlane_ios_custom_build_and_deploy.yaml (hunk 7 above)
Comment (DDavidPrime) [reply packages/rx_bloc_cli/mason_templates/bricks/feature_cicd_fastlane/__brick__/{{#cicd_github}}.github{{/cicd_github}}/workflows/fastlane_ios_custom_build_and_deploy.yaml]: After some look at the code history, it turns out this came from develop. Those changes were made intentionally so closing this.

File: packages/rx_bloc_cli/mason_templates/bricks/feature_cicd_fastlane/__brick__/{{#cicd_github}}.github{{/cicd_github}}/workflows/fastlane_android_custom_build_and_deploy.yaml (hunk 3 above)
Comment (DDavidPrime) [reply packages/rx_bloc_cli/mason_templates/bricks/feature_cicd_fastlane/__brick__/{{#cicd_github}}.github{{/cicd_github}}/workflows/fastlane_android_custom_build_and_deploy.yaml]: As it turns out, the change is actually merged from `develop` where we removed the caching for android/ios build phases. We discussed this with @StanevPrime and agreed that it does not bring any benefits to the build/deploy workflows as it only adds additional minutes for caching (which is not used when in combination with tags).

File: examples/todoapp/lib/base/common_services/deep_link_service.dart (hunk 4 above)
Comment (nikolay-vasilev-prime) [reply examples/todoapp/lib/base/common_services/deep_link_service.dart]: Done

File: examples/todoapp/lib/base/data_sources/local/profile_local_data_source.dart (hunk 5 above)
Comment (nikolay-vasilev-prime) [reply examples/todoapp/lib/base/data_sources/local/profile_local_data_source.dart]: Done

File: packages/rx_bloc_cli/mason_templates/bricks/feature_cicd_fastlane/__brick__/{{#cicd_github}}.github{{/cicd_github}}/workflows/fastlane_android_custom_build_and_deploy.yaml (hunk 3 above)
Comment (StanevPrime) [reply packages/rx_bloc_cli/mason_templates/bricks/feature_cicd_fastlane/__brick__/{{#cicd_github}}.github{{/cicd_github}}/workflows/fastlane_android_custom_build_and_deploy.yaml]: The change has been reverted.

**Comment by nikolay-vasilev-prime** [comment]
Closing the pr, because github doesn't show changes of deleted files which exist in the develop branch.
A new PR is created -> https://github.com/Prime-Holding/rx_bloc/pull/735

**Pull request closed without merging**